### PR TITLE
Capstone v6 now supports Debian Packaging

### PR DESCRIPTION
**Pull request by AndrewQuijano**
Please see PR #2569, this is duplicating the workflow for Capstone v6 Branch

Past work:
https://github.com/capstone-engine/capstone/pull/2543
https://github.com/capstone-engine/capstone/pull/2521

### CODE REVIEW

**Comment by AndrewQuijano** [comment]
Here is the output, slightly different from v5 as it has cmake stuff still, but everything works as intended

```
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:35 ..
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 usr

extract/usr:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 ..
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 bin
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 include
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 lib
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 share

extract/usr/bin:
total 22044
drwxrwxrwx 1 andrew andrew     4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew     4096 Dec  7 12:33 ..
-rwxrwxrwx 1 andrew andrew 22569664 Dec  7 12:33 cstool

extract/usr/include:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 ..
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 capstone

extract/usr/include/capstone:
total 1108
drwxrwxrwx 1 andrew andrew   4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew   4096 Dec  7 12:33 ..
-rwxrwxrwx 1 andrew andrew 155304 Dec  7 12:01 aarch64.h
-rwxrwxrwx 1 andrew andrew   6450 Dec  7 12:01 alpha.h
-rwxrwxrwx 1 andrew andrew  42392 Dec  7 12:01 arm.h
-rwxrwxrwx 1 andrew andrew 208200 Dec  7 12:01 arm64.h
-rwxrwxrwx 1 andrew andrew   4365 Dec  7 12:01 bpf.h
-rwxrwxrwx 1 andrew andrew  39026 Dec  7 12:01 capstone.h
-rwxrwxrwx 1 andrew andrew   1801 Dec  7 12:01 cs_operand.h
-rwxrwxrwx 1 andrew andrew   4574 Nov 28 13:27 evm.h
-rwxrwxrwx 1 andrew andrew  10066 Dec  7 12:01 hppa.h
-rwxrwxrwx 1 andrew andrew  69835 Dec  7 12:01 loongarch.h
-rwxrwxrwx 1 andrew andrew  12856 Dec  7 12:01 m680x.h
-rwxrwxrwx 1 andrew andrew  14467 Dec  7 12:01 m68k.h
-rwxrwxrwx 1 andrew andrew  53559 Dec  7 12:01 mips.h
-rwxrwxrwx 1 andrew andrew   5990 Dec  7 12:01 mos65xx.h
-rwxrwxrwx 1 andrew andrew   4132 Nov 28 13:27 platform.h
-rwxrwxrwx 1 andrew andrew  97926 Dec  7 12:01 ppc.h
-rwxrwxrwx 1 andrew andrew  13919 Dec  7 12:01 riscv.h
-rwxrwxrwx 1 andrew andrew   8091 Dec  7 12:01 sh.h
-rwxrwxrwx 1 andrew andrew  11917 Dec  7 12:01 sparc.h
-rwxrwxrwx 1 andrew andrew  68749 Dec  7 12:01 systemz.h
-rwxrwxrwx 1 andrew andrew 113838 Dec  7 12:01 systemz_compatibility.h
-rwxrwxrwx 1 andrew andrew   8621 Nov 28 13:27 tms320c64x.h
-rwxrwxrwx 1 andrew andrew  13570 Dec  7 12:01 tricore.h
-rwxrwxrwx 1 andrew andrew   6441 Nov 28 13:27 wasm.h
-rwxrwxrwx 1 andrew andrew  45064 Dec  7 12:01 x86.h
-rwxrwxrwx 1 andrew andrew   5154 Nov 28 13:27 xcore.h
-rwxrwxrwx 1 andrew andrew  48347 Dec  7 12:01 xtensa.h

extract/usr/lib:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 ..
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 pkgconfig
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:34 x86_64-linux-gnu

extract/usr/lib/pkgconfig:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 ..

extract/usr/lib/x86_64-linux-gnu:
total 45836
drwxrwxrwx 1 andrew andrew     4096 Dec  7 12:34 .
drwxrwxrwx 1 andrew andrew     4096 Dec  7 12:33 ..
drwxrwxrwx 1 andrew andrew     4096 Dec  7 12:33 cmake
-rwxrwxrwx 1 andrew andrew 24360514 Dec  7 12:33 libcapstone.a
lrwxrwxrwx 1 andrew andrew       16 Dec  7 12:33 libcapstone.so -> libcapstone.so.6
lrwxrwxrwx 1 andrew andrew       20 Dec  7 12:33 libcapstone.so.6 -> libcapstone.so.6.0.1
-rwxrwxrwx 1 andrew andrew 22569944 Dec  7 12:33 libcapstone.so.6.0.1
drwxrwxrwx 1 andrew andrew     4096 Dec  7 12:33 pkgconfig

extract/usr/lib/x86_64-linux-gnu/cmake:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:34 ..
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 capstone

extract/usr/lib/x86_64-linux-gnu/cmake/capstone:
total 20
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 ..
-rwxrwxrwx 1 andrew andrew 2878 Dec  7 12:32 capstone-config-version.cmake
-rwxrwxrwx 1 andrew andrew 1477 Dec  7 12:32 capstone-config.cmake
-rwxrwxrwx 1 andrew andrew 1916 Dec  7 12:32 capstone-targets-release.cmake
-rwxrwxrwx 1 andrew andrew 4762 Dec  7 12:32 capstone-targets.cmake

extract/usr/lib/x86_64-linux-gnu/pkgconfig:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:34 ..
-rwxrwxrwx 1 andrew andrew  364 Dec  7 12:32 capstone.pc

extract/usr/share:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 ..
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 doc

extract/usr/share/doc:
total 0
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 ..
drwxrwxrwx 1 andrew andrew 4096 Dec  7 12:33 libcapstone-dev

extract/usr/share/doc/libcapstone-dev:
total 76
drwxrwxrwx 1 andrew andrew  4096 Dec  7 12:33 .
drwxrwxrwx 1 andrew andrew  4096 Dec  7 12:33 ..
-rwxrwxrwx 1 andrew andrew  3352 Dec  7 12:01 CREDITS.TXT
-rwxrwxrwx 1 andrew andrew 56302 Dec  7 12:01 ChangeLog
-rwxrwxrwx 1 andrew andrew  3998 Dec  7 11:53 HACK.TXT
-rwxrwxrwx 1 andrew andrew  1687 Dec  7 11:53 LICENSE.TXT
-rwxrwxrwx 1 andrew andrew  3103 Dec  7 12:01 README.md
-rwxrwxrwx 1 andrew andrew   919 Dec  7 12:01 SPONSORS.TXT
```

capstone.pc
image

control file
image

File: LICENSE.TXT
Comment (Rot127) [inline LICENSE.TXT:1]: Remove this one please. The license files are in: `LICENSES/`.

File: HACK.TXT
Comment (Rot127) [inline HACK.TXT:1]: Remove this one please. It was replaced with `CONTRIBUTING.md`. But I don't think it gives any useful information for the package. So no need to include it at all.

File: packages/deb/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# SPDX-License-Identifier: MIT
```
Comment (Rot127) [inline packages/deb/Dockerfile:1]: If you want this under the MIT license, please add the license file with [reuse](https://pypi.org/project/reuse/):
```
reuse download MIT
```

File: packages/deb/README.md
Comment (Rot127) [inline packages/deb/README.md:1]: Build instructions would still be nice to have.

Please leave at least a little documentation here:
```
This is the deb package built with docker.
To build do:
etc. pp.
```

**Review by Rot127 — CHANGES_REQUESTED** [review]

**Comment by Rot127** [comment]
Ah, also. Could you please do a release on your fork with `6.0.0-Alpha8` or something to check if it works?
Then add the link here.

File: HACK.TXT
Comment (AndrewQuijano) [reply HACK.TXT:1]: Done, ill remove HACK.TXT

File: LICENSE.TXT
Comment (AndrewQuijano) [reply LICENSE.TXT:1]: Done, ill copy the folder over then

**Comment by AndrewQuijano** [comment]
Here is the package

https://github.com/AndrewQuijano/capstone/releases/tag/6.0.0-Alpha10

Licenses copied over correctly too
image

File: packages/deb/README.md [hunk 2]
```diff
@@ -1,7 +1,24 @@
-Incomplete Debian package implementation.
-It can be used to generate an easily installable Capstone, but misses a lot of
-mandatory things to add it in the Debian repos (`debian/control` is incomplete, no dependencies added etc.).
+# Capstone Docker packaging
 
-You can build the package by dispatching the `Build Debian Package` workflow or executing the commands in the `Dockerfile`.
-It assumes the current commit is tagged and `"$(git describe --tags --abbrev=0)"` returns a valid version number.
-The package is uploaded as artifact.
+This assumes your working directory is in the `packages/deb/` directory. 
+To build a Debian package for capstone, run the script, where `<tag-name>` is going to be the version
```
Comment (Rot127) [inline packages/deb/README.md]:
```suggestion
To build a Debian package for Capstone, run the script, where `<tag-name>` is going to be the version 
```

File: packages/deb/README.md [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+# Capstone Docker packaging
 
-You can build the package by dispatching the `Build Debian Package` workflow or executing the commands in the `Dockerfile`.
-It assumes the current commit is tagged and `"$(git describe --tags --abbrev=0)"` returns a valid version number.
-The package is uploaded as artifact.
+This assumes your working directory is in the `packages/deb/` directory. 
+To build a Debian package for capstone, run the script, where `<tag-name>` is going to be the version 
+attached to both the `capstone.pc` file and the Debian package itself. The version is expected to be compliant with Debian versioning (a major/minor/patch), e. g. `5.0.4`. Debian versions can also support values to indicate pre-release, e. g. `6.0.0-Alpha1`.
+
+Note if a value such as `6.0.0-Alpha1` is provided, the major/minor/patch number is extracted for `capstone.pc`, which would have version `6.0.0`, but the Debian package would have the full version name on the control file.
```
Comment (Rot127) [inline packages/deb/README.md]:
```suggestion
**Note**: if a value such as `6.0.0-Alpha1` is provided, the major/minor/patch number is extracted for `capstone.pc`, which would have version `6.0.0`, but the Debian package would have the full version name on the control file.

**Note**: Currently the package is hard coded to the `amd64` architecture. Independently on what machine you built it. Also see [issue #2537](https://github.com/capstone-engine/capstone/issues/2537).
```

**Review by Rot127 — APPROVED** [review]
Good job!
Please just check the last two things. Then we are good to go IMO.

File: LICENSES/MIT.txt
Comment (kabeor) [inline LICENSES/MIT.txt:1]: Thanks for your contribution. Please rename this file to LICENSE_DEB_PACKAGE.txt. In addition, please indicate the effective scope of MIT LICENSE, which is packages/deb.

**Comment by AndrewQuijano** [comment]
@kabeor updated the license, please merge the debian package for v5 and v6, and create a 5.0.4 release

**Review by kabeor — APPROVED** [review]
Thank you. Merge now.

**Pull request merged**